In [1]:
#necessary libraries and modules to import

import sqlite3
import os
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from glob import glob

In [2]:
#changing the display settings in dataframes to only format floats and display columns

pd.options.display.float_format = '{:.2f}'.format
pd.set_option('display.max_columns', None)
pd.set_option('max_rows', 200)

In [3]:
#converts tsv files

tsv_file = './zippedData/rt.movie_info.tsv.gz'
convert = pd.read_table(tsv_file, delimiter='\t')
convert.to_csv('./zippedData/rt.movie_info.csv.gz')

tsv_file_r = './zippedData/rt.reviews.tsv.gz'
convert_r = pd.read_table(tsv_file_r, delimiter='\t', encoding = 'unicode_escape')
convert_r.to_csv('./zippedData/rt.reviews.csv.gz')

In [4]:
#importing batch data using os

csv_files = glob("./zippedData/*.csv.gz")
csv_files

['./zippedData\\bom.movie_gross.csv.gz',
 './zippedData\\imdb.name.basics.csv.gz',
 './zippedData\\imdb.title.akas.csv.gz',
 './zippedData\\imdb.title.basics.csv.gz',
 './zippedData\\imdb.title.crew.csv.gz',
 './zippedData\\imdb.title.principals.csv.gz',
 './zippedData\\rt.movie_info.csv.gz',
 './zippedData\\rt.reviews.csv.gz',
 './zippedData\\tmdb.movies.csv.gz',
 './zippedData\\tn.movie_budgets.csv.gz']

In [5]:
# cleaning the filenames

csv_files_dict = {} #create a dictionary of datasets
for filename in csv_files: #create a for loop to batch clean files
    filename_cleaned = os.path.basename(filename).replace(".csv", "").replace(".", "_") #remove .csv file extensions
    filename_df = pd.read_csv(filename, index_col=0, encoding='utf-8') 
    csv_files_dict[filename_cleaned] = filename_df #load .csv file as dataframe using col 1 as index and encode in utf-8 and save as the cleaned filename

In [6]:
print(csv_files_dict.keys())

dict_keys(['bom_movie_gross_gz', 'imdb_name_basics_gz', 'imdb_title_akas_gz', 'imdb_title_basics_gz', 'imdb_title_crew_gz', 'imdb_title_principals_gz', 'rt_movie_info_gz', 'rt_reviews_gz', 'tmdb_movies_gz', 'tn_movie_budgets_gz'])


In [48]:
bom_movie_gross_df = csv_files_dict['bom_movie_gross_gz']
imdb_name_basics_df = csv_files_dict['imdb_name_basics_gz']
imdb_title_akas_df = csv_files_dict['imdb_title_akas_gz']
imdb_title_basics_df = csv_files_dict['imdb_title_basics_gz']
imdb_title_crew_df = csv_files_dict['imdb_title_crew_gz']
imdb_title_principals_df = csv_files_dict['imdb_title_principals_gz']
rotten_movie_info_df = csv_files_dict['rt_movie_info_gz']
rotten_movie_reviews_df = csv_files_dict['rt_reviews_gz']
tmdb_movies_df = csv_files_dict['tmdb_movies_gz']
tmdb_genre_ids_df = pd.read_json('tmdb_genre_ids.json')
tmdb_languages_df = pd.read_json('tmdb_languages.json')
tn_movie_budgets_df = csv_files_dict['tn_movie_budgets_gz']

In [8]:
bom_movie_gross_df

studio  domestic_gross  \
title                                                                     
Toy Story 3                                          BV    415000000.00   
Alice in Wonderland (2010)                           BV    334200000.00   
Harry Potter and the Deathly Hallows Part 1          WB    296000000.00   
Inception                                            WB    292600000.00   
Shrek Forever After                                P/DW    238700000.00   
...                                                 ...             ...   
The Quake                                         Magn.         6200.00   
Edward II (2018 re-release)                          FM         4800.00   
El Pacto                                           Sony         2500.00   
The Swan                                     Synergetic         2400.00   
An Actor Prepares                                 Grav.         1700.00   

                                            foreign_gross  year  
title                                                            
Toy Story 3                                     652000000  2010  
Alice in Wonderland (2010)                      691300000  2010  
Harry Potter and the Deathly Hallows Part 1     664300000  2010  
Inception                                       535700000  2010  
Shrek Forever After                             513900000  2010  
...                                                   ...   ...  
The Quake                                             NaN  2018  
Edward II (2018 re-release)                           NaN  2018  
El Pacto                                              NaN  2018  
The Swan                                              NaN  2018  
An Actor Prepares                                     NaN  2018  

[3387 rows x 4 columns]

In [9]:
rotten_movie_reviews_df.head()

id                                             review rating   fresh  \
0   3  A distinctly gallows take on contemporary fina...    3/5   fresh   
1   3  It's an allegory in search of a meaning that n...    NaN  rotten   
2   3  ... life lived in a bubble in financial dealin...    NaN   fresh   
3   3  Continuing along a line introduced in last yea...    NaN   fresh   
4   3             ... a perverse twist on neorealism...     NaN   fresh   

           critic  top_critic         publisher               date  
0      PJ Nabarro           0   Patrick Nabarro  November 10, 2018  
1  Annalee Newitz           0           io9.com       May 23, 2018  
2    Sean Axmaker           0  Stream on Demand    January 4, 2018  
3   Daniel Kasman           0              MUBI  November 16, 2017  
4             NaN           0      Cinema Scope   October 12, 2017

In [10]:
tmdb_movies_df.iloc[50:100,:]

genre_ids     id original_language  \
50             [12, 28, 18]  23759                en   
51             [28, 35, 80]  27581                en   
52             [80, 18, 53]  23168                en   
53     [28, 37, 18, 14, 53]  20533                en   
54            [12, 28, 878]  20526                en   
55      [12, 14, 28, 10749]   9543                en   
56          [35, 16, 10751]   9994                en   
57                 [14, 18]   7980                en   
58                     [18]    705                en   
59             [18, 12, 37]  44264                en   
60             [53, 80, 27]  43947                en   
61                 [28, 80]  23483                en   
62            [35, 878, 16]  51888                en   
63           [28, 9648, 53]  27576                en   
64   [10770, 16, 14, 10751]  47626                en   
65             [28, 80, 53]  26389                en   
66           [28, 16, 9648]  40662                en   
67      [16, 12, 14, 10751]  42949                en   
68          [35, 10751, 14]  23023                en   
69             [28, 35, 53]  23631                en   
70    [80, 18, 9648, 10749]  25376                es   
71                     [35]  35056                en   
72                 [35, 18]  41733                en   
73        [28, 878, 53, 80]  31867                en   
74                 [16, 28]  43641                en   
75                     [16]  28874                ja   
76      [28, 12, 18, 10751]  38575                en   
77              [35, 10751]  33217                en   
78           [9648, 27, 28]  29427                en   
79       [28, 53, 80, 9648]  33613                sv   
80              [35, 10749]  39451                en   
81     [28, 12, 35, 80, 53]  39514                en   
82       [18, 10749, 10752]  22971                en   
83                     [27]  22894                en   
84        [12, 14, 16, 878]  37933                ja   
85              [10749, 35]  25195                en   
86          [35, 10749, 18]  38408                en   
87   [18, 28, 53, 80, 9648]  24253                sv   
88                 [18, 53]  38363                en   
89       [35, 10402, 10770]  44244                en   
90  [10752, 28, 12, 18, 53]  22972                en   
91             [28, 12, 16]  30061                en   
92              [18, 10749]  23169                en   
93             [12, 18, 53]  44115                en   
94       [9648, 14, 35, 12]  47533                de   
95                 [35, 27]  43593                en   
96         [12, 18, 28, 14]  37861                en   
97       [80, 18, 9648, 53]  12201                en   
98       [14, 12, 28, 9648]  35552                fr   
99              [35, 10749]  32856                en   

                                     original_title  popularity release_date  \
50                                        Centurion       13.55   2010-02-15   
51                                   The Other Guys       13.51   2010-08-06   
52                                         The Town       13.48   2010-09-17   
53                                        Jonah Hex       13.47   2010-06-18   
54                                     TRON: Legacy       13.46   2010-12-10   
55              Prince of Persia: The Sands of Time       13.44   2010-05-27   
56                        The Great Mouse Detective       13.35   1986-07-02   
57                                 The Lovely Bones       13.29   2010-01-15   
58                                    All About Eve       13.16   2000-10-06   
59                                        True Grit       13.12   2010-12-22   
60                             I Spit on Your Grave       13.10   2010-10-08   
61                                         Kick-Ass       13.08   2010-04-16   
62             Robot Chicken: Star Wars Episode III       13.06   2010-12-19   
63                                       

In [11]:
tmdb_movies_df.shape

(26517, 9)

In [12]:
tmdb_movies_df['genre_ids'] = tmdb_movies_df['genre_ids'].astype(str)

In [13]:
tmdb_movies_df['genre_ids'] = tmdb_movies_df['genre_ids'].str.strip('[]')
#tmdb_movies_df['genre_ids'] = tmdb_movies_df['genre_ids'].str.strip('')

In [14]:
tmdb_movies_df

genre_ids      id original_language  \
0          12, 14, 10751   12444                en   
1      14, 12, 16, 10751   10191                en   
2            12, 28, 878   10138                en   
3          16, 35, 10751     862                en   
4            28, 878, 12   27205                en   
...                  ...     ...               ...   
26512             27, 18  488143                en   
26513             18, 53  485975                en   
26514         14, 28, 12  381231                en   
26515      10751, 12, 28  366854                en   
26516             53, 27  309885                en   

                                     original_title  popularity release_date  \
0      Harry Potter and the Deathly Hallows: Part 1       33.53   2010-11-19   
1                          How to Train Your Dragon       28.73   2010-03-26   
2                                        Iron Man 2       28.52   2010-05-07   
3                                         Toy Story       28.00   1995-11-22   
4                                         Inception       27.92   2010-07-16   
...                                             ...         ...          ...   
26512                         Laboratory Conditions        0.60   2018-10-13   
26513                               _EXHIBIT_84xxx_        0.60   2018-05-01   
26514                                  The Last One        0.60   2018-10-01   
26515                                  Trailer Made        0.60   2018-06-22   
26516                                    The Church        0.60   2018-10-05   

                                              title  vote_average  vote_count  
0      Harry Potter and the Deathly Hallows: Part 1          7.70       10788  
1                          How to Train Your Dragon          7.70        7610  
2                                        Iron Man 2          6.80       12368  
3                                         Toy Story          7.90       10174  
4                                         Inception          8.30       22186  
...                                             ...           ...         ...  
26512                         Laboratory Conditions          0.00           1  
26513                               _EXHIBIT_84xxx_          0.00           1  
26514                                  The Last One          0.00           1  
26515                                  Trailer Made          0.00           1  
26516                                    The Church          0.00           1  

[26517 rows x 9 columns]

In [15]:
#split the genre column to individual genre columns
#use a lambda function to split the genre list at the comma

tmdb_movies_df['genre_ids'] = tmdb_movies_df['genre_ids'].apply(lambda x: x.split(", ") if x else x)

tmdb_movies_df.head()

genre_ids     id original_language  \
0      [12, 14, 10751]  12444                en   
1  [14, 12, 16, 10751]  10191                en   
2        [12, 28, 878]  10138                en   
3      [16, 35, 10751]    862                en   
4        [28, 878, 12]  27205                en   

                                 original_title  popularity release_date  \
0  Harry Potter and the Deathly Hallows: Part 1       33.53   2010-11-19   
1                      How to Train Your Dragon       28.73   2010-03-26   
2                                    Iron Man 2       28.52   2010-05-07   
3                                     Toy Story       28.00   1995-11-22   
4                                     Inception       27.92   2010-07-16   

                                          title  vote_average  vote_count  
0  Harry Potter and the Deathly Hallows: Part 1          7.70       10788  
1                      How to Train Your Dragon          7.70        7610  
2                                    Iron Man 2          6.80       12368  
3                                     Toy Story          7.90       10174  
4                                     Inception          8.30       22186

In [16]:
#create a set (unordered) of all genres (no duplicate values) using a for loop

all_genres = set()
for genres in tmdb_movies_df['genre_ids']: #for all genres in the move_details_df 'genres' column
    if genres: #if in genres
        all_genres.update(genres) #add to the set called all_genres

In [17]:
all_genres

{'10402',
 '10749',
 '10751',
 '10752',
 '10770',
 '12',
 '14',
 '16',
 '18',
 '27',
 '28',
 '35',
 '36',
 '37',
 '53',
 '80',
 '878',
 '9648',
 '99'}

In [18]:
#add new genre cols from all_genres set and fill with zero values using numpy with a for loop; 

for genre in all_genres: #for genre in all_genres set
    tmdb_movies_df[genre] = np.zeros(shape=tmdb_movies_df.shape[0]) 
        #create new genre column such that it is filled with zeroes 
        #with a shape equal to first dataframe col
                                
tmdb_movies_df.head() #return first 5 rows of dataframe

genre_ids     id original_language  \
0      [12, 14, 10751]  12444                en   
1  [14, 12, 16, 10751]  10191                en   
2        [12, 28, 878]  10138                en   
3      [16, 35, 10751]    862                en   
4        [28, 878, 12]  27205                en   

                                 original_title  popularity release_date  \
0  Harry Potter and the Deathly Hallows: Part 1       33.53   2010-11-19   
1                      How to Train Your Dragon       28.73   2010-03-26   
2                                    Iron Man 2       28.52   2010-05-07   
3                                     Toy Story       28.00   1995-11-22   
4                                     Inception       27.92   2010-07-16   

                                          title  vote_average  vote_count  \
0  Harry Potter and the Deathly Hallows: Part 1          7.70       10788   
1                      How to Train Your Dragon          7.70        7610   
2                                    Iron Man 2          6.80       12368   
3                                     Toy Story          7.90       10174   
4                                     Inception          8.30       22186   

    16   28   14  10749   18  878  10751  10402   36  10752   80   35  9648  \
0 0.00 0.00 0.00   0.00 0.00 0.00   0.00   0.00 0.00   0.00 0.00 0.00  0.00   
1 0.00 0.00 0.00   0.00 0.00 0.00   0.00   0.00 0.00   0.00 0.00 0.00  0.00   
2 0.00 0.00 0.00   0.00 0.00 0.00   0.00   0.00 0.00   0.00 0.00 0.00  0.00   
3 0.00 0.00 0.00   0.00 0.00 0.00   0.00   0.00 0.00   0.00 0.00 0.00  0.00   
4 0.00 0.00 0.00   0.00 0.00 0.00   0.00   0.00 0.00   0.00 0.00 0.00  0.00   

    37   12   53  10770   27   99  
0 0.00 0.00 0.00   0.00 0.00 0.00  
1 0.00 0.00 0.00   0.00 0.00 0.00  
2 0.00 0.00 0.00   0.00 0.00 0.00  
3 0.00 0.00 0.00   0.00 0.00 0.00  
4 0.00 0.00 0.00   0.00 0.00 0.00

In [19]:
#set the genre value to 1 if the film is of that genre with a nested for loop

for index, row in tmdb_movies_df.iterrows(): #for index, row position in dataframe, iterating throw rows
    if row['genre_ids']: #if in genres col
        for genre in row['genre_ids']: #for genre in genres col
            tmdb_movies_df.loc[index, genre] = 1 #set the value of each matching index/genre position to 1

tmdb_movies_df.head() #return first 5 rows of dataframe

genre_ids     id original_language  \
0      [12, 14, 10751]  12444                en   
1  [14, 12, 16, 10751]  10191                en   
2        [12, 28, 878]  10138                en   
3      [16, 35, 10751]    862                en   
4        [28, 878, 12]  27205                en   

                                 original_title  popularity release_date  \
0  Harry Potter and the Deathly Hallows: Part 1       33.53   2010-11-19   
1                      How to Train Your Dragon       28.73   2010-03-26   
2                                    Iron Man 2       28.52   2010-05-07   
3                                     Toy Story       28.00   1995-11-22   
4                                     Inception       27.92   2010-07-16   

                                          title  vote_average  vote_count  \
0  Harry Potter and the Deathly Hallows: Part 1          7.70       10788   
1                      How to Train Your Dragon          7.70        7610   
2                                    Iron Man 2          6.80       12368   
3                                     Toy Story          7.90       10174   
4                                     Inception          8.30       22186   

    16   28   14  10749   18  878  10751  10402   36  10752   80   35  9648  \
0 0.00 0.00 1.00   0.00 0.00 0.00   1.00   0.00 0.00   0.00 0.00 0.00  0.00   
1 1.00 0.00 1.00   0.00 0.00 0.00   1.00   0.00 0.00   0.00 0.00 0.00  0.00   
2 0.00 1.00 0.00   0.00 0.00 1.00   0.00   0.00 0.00   0.00 0.00 0.00  0.00   
3 1.00 0.00 0.00   0.00 0.00 0.00   1.00   0.00 0.00   0.00 0.00 1.00  0.00   
4 0.00 1.00 0.00   0.00 0.00 1.00   0.00   0.00 0.00   0.00 0.00 0.00  0.00   

    37   12   53  10770   27   99  
0 0.00 1.00 0.00   0.00 0.00 0.00  
1 0.00 1.00 0.00   0.00 0.00 0.00  
2 0.00 1.00 0.00   0.00 0.00 0.00  
3 0.00 0.00 0.00   0.00 0.00 0.00  
4 0.00 1.00 0.00   0.00 0.00 0.00

In [20]:
tmdb_genre_ids_df

genres
0             {'id': 28, 'name': 'Action'}
1          {'id': 12, 'name': 'Adventure'}
2          {'id': 16, 'name': 'Animation'}
3             {'id': 35, 'name': 'Comedy'}
4              {'id': 80, 'name': 'Crime'}
5        {'id': 99, 'name': 'Documentary'}
6              {'id': 18, 'name': 'Drama'}
7          {'id': 10751, 'name': 'Family'}
8            {'id': 14, 'name': 'Fantasy'}
9            {'id': 36, 'name': 'History'}
10            {'id': 27, 'name': 'Horror'}
11          {'id': 10402, 'name': 'Music'}
12         {'id': 9648, 'name': 'Mystery'}
13        {'id': 10749, 'name': 'Romance'}
14  {'id': 878, 'name': 'Science Fiction'}
15       {'id': 10770, 'name': 'TV Movie'}
16          {'id': 53, 'name': 'Thriller'}
17            {'id': 10752, 'name': 'War'}
18           {'id': 37, 'name': 'Western'}

In [21]:
keys = tmdb_genre_ids_df.genres.iloc[0].keys() #Get dictionary keys

#Keep track of columns we make for subsequent preview
genre_cols = []

#Create a new feature for each of these keys
for key in keys:
    new_col = 'genres_{}'.format(key) #Create new column name
    tmdb_genre_ids_df[new_col] = tmdb_genre_ids_df.genres.map(lambda x: x[key]) #Create a new column
    genre_cols.append(new_col)
    
tmdb_genre_ids_df[genre_cols]

genres_id      genres_name
0          28           Action
1          12        Adventure
2          16        Animation
3          35           Comedy
4          80            Crime
5          99      Documentary
6          18            Drama
7       10751           Family
8          14          Fantasy
9          36          History
10         27           Horror
11      10402            Music
12       9648          Mystery
13      10749          Romance
14        878  Science Fiction
15      10770         TV Movie
16         53         Thriller
17      10752              War
18         37          Western

In [22]:
tmdb_genre_cols = tmdb_genre_ids_df[genre_cols]
tmdb_genre_cols

genres_id      genres_name
0          28           Action
1          12        Adventure
2          16        Animation
3          35           Comedy
4          80            Crime
5          99      Documentary
6          18            Drama
7       10751           Family
8          14          Fantasy
9          36          History
10         27           Horror
11      10402            Music
12       9648          Mystery
13      10749          Romance
14        878  Science Fiction
15      10770         TV Movie
16         53         Thriller
17      10752              War
18         37          Western

In [23]:
genres_dict = dict(zip(tmdb_genre_cols['genres_id'], tmdb_genre_cols['genres_name']))

genres_dict

{28: 'Action',
 12: 'Adventure',
 16: 'Animation',
 35: 'Comedy',
 80: 'Crime',
 99: 'Documentary',
 18: 'Drama',
 10751: 'Family',
 14: 'Fantasy',
 36: 'History',
 27: 'Horror',
 10402: 'Music',
 9648: 'Mystery',
 10749: 'Romance',
 878: 'Science Fiction',
 10770: 'TV Movie',
 53: 'Thriller',
 10752: 'War',
 37: 'Western'}

In [24]:
genres_dict = {str(k): v for k, v in genres_dict.items()}

In [25]:
tmdb_movies_genres_df = tmdb_movies_df.rename(columns=genres_dict)
tmdb_movies_genres_df

genre_ids      id original_language  \
0          [12, 14, 10751]   12444                en   
1      [14, 12, 16, 10751]   10191                en   
2            [12, 28, 878]   10138                en   
3          [16, 35, 10751]     862                en   
4            [28, 878, 12]   27205                en   
...                    ...     ...               ...   
26512             [27, 18]  488143                en   
26513             [18, 53]  485975                en   
26514         [14, 28, 12]  381231                en   
26515      [10751, 12, 28]  366854                en   
26516             [53, 27]  309885                en   

                                     original_title  popularity release_date  \
0      Harry Potter and the Deathly Hallows: Part 1       33.53   2010-11-19   
1                          How to Train Your Dragon       28.73   2010-03-26   
2                                        Iron Man 2       28.52   2010-05-07   
3                                         Toy Story       28.00   1995-11-22   
4                                         Inception       27.92   2010-07-16   
...                                             ...         ...          ...   
26512                         Laboratory Conditions        0.60   2018-10-13   
26513                               _EXHIBIT_84xxx_        0.60   2018-05-01   
26514                                  The Last One        0.60   2018-10-01   
26515                                  Trailer Made        0.60   2018-06-22   
26516                                    The Church        0.60   2018-10-05   

                                              title  vote_average  vote_count  \
0      Harry Potter and the Deathly Hallows: Part 1          7.70       10788   
1                          How to Train Your Dragon          7.70        7610   
2                                        Iron Man 2          6.80       12368   
3                                         Toy Story          7.90       10174   
4                                         Inception          8.30       22186   
...                                             ...           ...         ...   
26512                         Laboratory Conditions          0.00           1   
26513                               _EXHIBIT_84xxx_          0.00           1   
26514                                  The Last One          0.00           1   
26515                                  Trailer Made          0.00           1   
26516                                    The Church          0.00           1   

       Animation  Action  Fantasy  Romance  Drama  Science Fiction  Family  \
0           0.00    0.00     1.00     0.00   0.00             0.00    1.00   
1           1.00    0.00     1.00     0.00   0.00             0.00    1.00   
2           0.00    1.00     0.00     0.00   0.00             1.00    0.00   
3           1.00    0.00     0.00     0.00   0.00             0.00    1.00   
4           0.00    1.00     0.00     0.00   0.00             1.00    0.00   
...          ...     ...      ...      ...    ...              ...     ...   
26512       0.00    0.00     0.00     0.00   1.00             0.00    0.00   
26513       0.00    0.00     0.00     0.00   1.00             0.00    0.00   
26514       0.00    1.00     1.00     0.00   0.00             0.00    0.00   
26515       0.00    1.00     0.00     0.00   0.00             0.00    1.00   
26516       0.00    0.00     0.00     0.00   0.00             0.00    0.00   

       Music  History  War  Crime  Comedy  Mystery  Western  Adventure  \
0       0.00     0.00 0.00   0.00    0.00     0.00     0.00       1.00   
1       0.00     0.00 0.00   0.00    0.00     0.00     0.00       1.00   
2       0.00     0.00 0.00   0.00    0.00     0.00     0.00       1.00   
3       0.00     0.00 0.00   0.00    1.00     0.00     0.00       0.00   
4       0.00     0.00 0.00   0.00    0.00     0.00     0.00       1.00   
...      ...      ...  ...   

#tmdb_movies_df.rename(columns={tmdb_genre_cols['genres_id']:tmdb_genre_cols['genres_name']})
 

keys = tmdb_genre_cols['genres_id']
values = tmdb_genre_cols['genres_name']

genre_names_list = tmdb_genre_cols['genres_name'].tolist()
genre_names_set = set(genre_names_list)
genre_names_set

#add new genre cols from all_genres set and fill with zero values using numpy with a for loop; 

for genre in genre_names_set: #for genre in all_genres set
    tmdb_movies_df[genre] = np.zeros(shape=tmdb_movies_df.shape[0]) 
        #create new genre column such that it is filled with zeroes 
        #with a shape equal to first dataframe col
                                
tmdb_movies_df.head() #return first 5 rows of dataframe

In [26]:
imdb_title_crew_df.head(10)

directors  \
tconst                                      
tt0285252                       nm0899854   
tt0438973                             NaN   
tt0462036                       nm1940585   
tt0835418                       nm0151540   
tt0878654   nm0089502,nm2291498,nm2292011   
tt0879859                       nm2416460   
tt0996958                       nm2286991   
tt0999913                       nm0527109   
tt10003792                     nm10539228   
tt10005130                     nm10540239   

                                            writers  
tconst                                               
tt0285252                                 nm0899854  
tt0438973                       nm0175726,nm1802864  
tt0462036                                 nm1940585  
tt0835418                       nm0310087,nm0841532  
tt0878654                                 nm0284943  
tt0879859                                       NaN  
tt0996958                       nm2286991,nm2651190  
tt0999913   nm0527109,nm0329051,nm0001603,nm0930684  
tt10003792                               nm10539228  
tt10005130                     nm5482263,nm10540239

In [27]:
imdb_title_principals_df.head(10)

ordering     nconst  category       job                characters
tconst                                                                      
tt0111414         1  nm0246005     actor       NaN               ["The Man"]
tt0111414         2  nm0398271  director       NaN                       NaN
tt0111414         3  nm3739909  producer  producer                       NaN
tt0323808        10  nm0059247    editor       NaN                       NaN
tt0323808         1  nm3579312   actress       NaN          ["Beth Boothby"]
tt0323808         2  nm2694680     actor       NaN         ["Steve Thomson"]
tt0323808         3  nm0574615     actor       NaN  ["Sir Lachlan Morrison"]
tt0323808         4  nm0502652   actress       NaN   ["Lady Delia Morrison"]
tt0323808         5  nm0362736  director       NaN                       NaN
tt0323808         6  nm0811056  producer  producer                       NaN

In [28]:
imdb_title_akas_df.head(50)

ordering                                    title region language  \
title_id                                                                       
tt0369610        10                            Джурасик свят     BG       bg   
tt0369610        11                        Jurashikku warudo     JP      NaN   
tt0369610        12  Jurassic World: O Mundo dos Dinossauros     BR      NaN   
tt0369610        13                  O Mundo dos Dinossauros     BR      NaN   
tt0369610        14                           Jurassic World     FR      NaN   
tt0369610        15                           Jurassic World     GR      NaN   
tt0369610        16                           Jurassic World     IT      NaN   
tt0369610        17                            Jurski svijet     HR      NaN   
tt0369610        18                             Olam ha'Yura     IL       he   
tt0369610        19           Jurassic World: Mundo Jurásico     MX      NaN   
tt0369610         1          Jurassic World: Sauruste maailm     EE      NaN   
tt0369610        20                           Jurassic World     SE      NaN   
tt0369610        21                        Jurassic World 3D     US      NaN   
tt0369610        22                        Jurassic World 3D     DE      NaN   
tt0369610        23                        Jurassic World 3D    XWW      NaN   
tt0369610        24                   Juras laikmeta pasaule     LV      NaN   
tt0369610        25                          Mundo Jurássico     PT      NaN   
tt0369610        26                      Мир Юрского периода     RU      NaN   
tt0369610        27                              Jurský svět     CZ      NaN   
tt0369610        28                         Monde jurassique     CA       fr   
tt0369610        29                           Jurassic World     US      NaN   
tt0369610         2                                 Ebb Tide     US      NaN   
tt0369610        30           Jurassic World: Mundo Jurásico     CL      NaN   
tt0369610        31                           Jurassic World     RO      NaN   
tt0369610        32                   Juros periodo pasaulis     LT      NaN   
tt0369610        33                           Jurassic World     FI      NaN   
tt0369610        34           Jurassic World: Mundo jurásico     PE      NaN   
tt0369610        35                          Jurassic Park 4     TR       tr   
tt0369610        36                         Jurassic Park IV     US      NaN   
tt0369610        37                           Jurassic World     NO      NaN   
tt0369610        38                              Jurski svet     SI      NaN   
tt0369610        39                    Свiт Юрського перiоду     UA      NaN   
tt0369610         3                      The Gioi Khung Long     VN      NaN   
tt0369610        40                           Jurassic World     DK      NaN   
tt0369610        41                        Svet iz doba jure     RS      NaN   
tt0369610        42                              Jurský svet     SK      NaN   
tt0369610        43                           Jurassic World     TR       tr   
tt0369610        44                          Jurassic Park 4     US      NaN   
tt0369610        45                           Jurassic World    NaN      NaN   
tt0369610        46                           Jurassic World     ES      NaN   
tt0369610        47                           Jurassic World     HU      NaN   
tt0369610        48                          Park jurajski 4     PL      NaN   
tt0369610         4                        Yur dövrü Dünyasi     AZ      NaN   
tt0369610         5                  Iuruli periodis samyaro     GE       ka   
tt0369610         6                           Jurassic World     PL      NaN   
tt0369610         7                           Mundo Jurásico     UY      NaN   
tt0369610         8                        Jurassic World 3D     PL      NaN   
tt0369610         9          Jurassic World - Mundo Jurásico     AR      NaN   
tt0381957  

In [29]:
imdb_name_basics_df.head(10)

primary_name  birth_year  death_year  \
nconst                                                 
nm0061671  Mary Ellen Bauder         nan         nan   
nm0061865       Joseph Bauer         nan         nan   
nm0062070         Bruce Baum         nan         nan   
nm0062195       Axel Baumann         nan         nan   
nm0062798        Pete Baxter         nan         nan   
nm0062879     Ruel S. Bayani         nan         nan   
nm0063198              Bayou         nan         nan   
nm0063432      Stevie Be-Zet         nan         nan   
nm0063618          Jeff Beal     1963.00         nan   
nm0063750    Lindsay Beamish         nan         nan   

                                         primary_profession  \
nconst                                                        
nm0061671         miscellaneous,production_manager,producer   
nm0061865        composer,music_department,sound_department   
nm0062070                        miscellaneous,actor,writer   
nm0062195  camera_department,cinematographer,art_department   
nm0062798  production_designer,art_department,set_decorator   
nm0062879         director,production_manager,miscellaneous   
nm0063198                                             actor   
nm0063432                               composer,soundtrack   
nm0063618              composer,music_department,soundtrack   
nm0063750                             actress,miscellaneous   

                                  known_for_titles  
nconst                                              
nm0061671  tt0837562,tt2398241,tt0844471,tt0118553  
nm0061865  tt0896534,tt6791238,tt0287072,tt1682940  
nm0062070  tt1470654,tt0363631,tt0104030,tt0102898  
nm0062195  tt0114371,tt2004304,tt1618448,tt1224387  
nm0062798  tt0452644,tt0452692,tt3458030,tt2178256  
nm0062879  tt2590280,tt0352080,tt0216559,tt2057445  
nm0063198                      tt6579724,tt0093116  
nm0063432  tt3106212,tt0478239,tt0264917,tt1626606  
nm0063618  tt0183659,tt2545118,tt0384766,tt1856010  
nm0063750  tt0404826,tt0111756,tt0367027,tt1492842

In [30]:
imdb_title_basics_df.head(50)

primary_title  \
tconst                                                         
tt0063540                                          Sunghursh   
tt0066787                    One Day Before the Rainy Season   
tt0069049                         The Other Side of the Wind   
tt0069204                                    Sabse Bada Sukh   
tt0100275                           The Wandering Soap Opera   
tt0111414                                        A Thin Life   
tt0112502                                            Bigfoot   
tt0137204                                    Joe Finds Grace   
tt0139613                                         O Silêncio   
tt0144449                              Nema aviona za Zagreb   
tt0146592                                        Pál Adrienn   
tt0154039                               So Much for Justice!   
tt0159369                 Cooper and Hemingway: The True Gen   
tt0162942                       Children of the Green Dragon   
tt0170651                               T.G.M. - osvoboditel   
tt0176694                                 The Tragedy of Man   
tt0187902  How Huang Fei-hong Rescued the Orphan from the...   
tt0192528                                      Heaven & Hell   
tt0230212                                  The Final Journey   
tt0247643                   Los pájaros se van con la muerte   
tt0249516                                         Foodfight!   
tt0250404                                          Godfather   
tt0253093                                       Gangavataran   
tt0255820                                  Return to Babylon   
tt0262218                                  Akakis mogzauroba   
tt0262759                           Seven Jews from My Class   
tt0263814                                           On kadin   
tt0276568                                 To Chase a Million   
tt0277964               Prague: The Restless Heart of Europe   
tt0283440                                  Short Time Heroes   
tt0285252                                     Life's a Beach   
tt0285423                                           Abolição   
tt0293069                                         Dark Blood   
tt0293429                                      Mortal Kombat   
tt0297400                                          Snowblind   
tt0306058                                      Second Coming   
tt0311448                               Lives No Longer Ours   
tt0312305             Quantum Quest: A Cassini Space Odyssey   
tt0315642                                              Wazir   
tt0323808                                    The Wicker Tree   
tt0326592                                      The Overnight   
tt0326965                                        In My Sleep   
tt0328810                                            My Land   
tt0329539                                          Rice Girl   
tt0330811                                Regret Not Speaking   
tt0330987                            Tiden är en dröm, del 2   
tt0331312                                 This Wretched Life   
tt0331314                                    Bunyan and Babe   
tt0337692                                        On the Road   
tt0337882                                        Blind Sided   

                                              original_title  start_year  \
tconst                                                                     
tt0063540                                          Sunghursh        2013   
tt0066787                                    Ashad Ka Ek Din        2019   
tt0069049                         The Other Side of the Wind        2018   
tt0069204                                    Sabse Bada Sukh        2018   
tt0100275                              La Telenovela Errante        2017   
tt0111414                                        A Thin Life        2018   
tt0112502                                            Bigfoot        2017   
tt0137204                         

## The Numbers dataset

### tn_movie_budgets_df

In [31]:
type(tn_movie_budgets_df) #call type of variable

pandas.core.frame.DataFrame

In [32]:
tn_movie_budgets_df.shape #return the dimensions of the dataframe array

(5782, 5)

In [33]:
tn_movie_budgets_df.info() #return basic summary of dataframe

<class 'pandas.core.frame.DataFrame'>
Int64Index: 5782 entries, 1 to 82
Data columns (total 5 columns):
release_date         5782 non-null object
movie                5782 non-null object
production_budget    5782 non-null object
domestic_gross       5782 non-null object
worldwide_gross      5782 non-null object
dtypes: object(5)
memory usage: 271.0+ KB


In [34]:
tn_movie_budgets_df.isna().sum() #sum of all the NaN values in each col

release_date         0
movie                0
production_budget    0
domestic_gross       0
worldwide_gross      0
dtype: int64

In [35]:
tn_movie_budgets_df.head() #call first five rows

release_date                                        movie  \
id                                                              
1   Dec 18, 2009                                       Avatar   
2   May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
3    Jun 7, 2019                                 Dark Phoenix   
4    May 1, 2015                      Avengers: Age of Ultron   
5   Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

   production_budget domestic_gross worldwide_gross  
id                                                   
1       $425,000,000   $760,507,625  $2,776,345,279  
2       $410,600,000   $241,063,875  $1,045,663,875  
3       $350,000,000    $42,762,350    $149,762,350  
4       $330,600,000   $459,005,868  $1,403,013,963  
5       $317,000,000   $620,181,382  $1,316,721,747

In [36]:
#define function called convert_amt_to_int
#cast datatype of col as string [added to avoid error], remove $, and change datatype to int64

def convert_amt_to_int(df, col):
    df[col] = df[col].astype('str').str.replace("$", "").str.replace(",", "").astype('int64') 
    return df

In [37]:
#make a list of cols to batch-change datatype with a for loop

money_cols = ['production_budget', 'domestic_gross', 'worldwide_gross']

for col in money_cols: 
    tn_movie_budgets_df = convert_amt_to_int(tn_movie_budgets_df, col)

In [38]:
tn_movie_budgets_df.info() #return basic summary of dataframe to confirm datatype changes

<class 'pandas.core.frame.DataFrame'>
Int64Index: 5782 entries, 1 to 82
Data columns (total 5 columns):
release_date         5782 non-null object
movie                5782 non-null object
production_budget    5782 non-null int64
domestic_gross       5782 non-null int64
worldwide_gross      5782 non-null int64
dtypes: int64(3), object(2)
memory usage: 271.0+ KB


In [39]:
tn_movie_sort = tn_movie_budgets_df.sort_values(by=['worldwide_gross'], ascending=False)
tn_movie_sort

release_date                                 movie  production_budget  \
id                                                                          
1   Dec 18, 2009                                Avatar          425000000   
43  Dec 19, 1997                               Titanic          200000000   
6   Dec 18, 2015  Star Wars Ep. VII: The Force Awakens          306000000   
7   Apr 27, 2018                Avengers: Infinity War          300000000   
34  Jun 12, 2015                        Jurassic World          215000000   
..           ...                                   ...                ...   
75  Dec 31, 2005                       Insomnia Manica             500000   
74  Jul 17, 2012                       Girls Gone Dead             500000   
73   Apr 3, 2012                         Enter Nowhere             500000   
72  Dec 31, 2010                                Drones             500000   
69  Dec 12, 2008                The Kings of Appletown            7000000   

    domestic_gross  worldwide_gross  
id                                   
1        760507625       2776345279  
43       659363944       2208208395  
6        936662225       2053311220  
7        678815482       2048134200  
34       652270625       1648854864  
..             ...              ...  
75               0                0  
74               0                0  
73               0                0  
72               0                0  
69               0                0  

[5782 rows x 5 columns]

lang = tmdb_mgl_df.groupby(['original_language', 'original_title']).count()

lang

# tmdb_lang_df work

In [40]:
tmdb_lang_df = pd.read_csv('tmdb_lang.csv')
tmdb_lang_df

Unnamed: 0      id                                original_title  \
0               0   12444  Harry Potter and the Deathly Hallows: Part 1   
1               1   10191                      How to Train Your Dragon   
2               2   10138                                    Iron Man 2   
3               3     862                                     Toy Story   
4               4   27205                                     Inception   
...           ...     ...                                           ...   
26512       26512  488143                         Laboratory Conditions   
26513       26513  485975                               _EXHIBIT_84xxx_   
26514       26514  381231                                  The Last One   
26515       26515  366854                                  Trailer Made   
26516       26516  309885                                    The Church   

       popularity release_date                                         title  \
0           33.53   2010-11-19  Harry Potter and the Deathly Hallows: Part 1   
1           28.73   2010-03-26                      How to Train Your Dragon   
2           28.52   2010-05-07                                    Iron Man 2   
3           28.00   1995-11-22                                     Toy Story   
4           27.92   2010-07-16                                     Inception   
...           ...          ...                                           ...   
26512        0.60   2018-10-13                         Laboratory Conditions   
26513        0.60   2018-05-01                               _EXHIBIT_84xxx_   
26514        0.60   2018-10-01                                  The Last One   
26515        0.60   2018-06-22                                  Trailer Made   
26516        0.60   2018-10-05                                    The Church   

       vote_average  vote_count  Comedy  Drama  Mystery  Western  Thriller  \
0              7.70       10788    0.00   0.00     0.00     0.00      0.00   
1              7.70        7610    0.00   0.00     0.00     0.00      0.00   
2              6.80       12368    0.00   0.00     0.00     0.00      0.00   
3              7.90       10174    1.00   0.00     0.00     0.00      0.00   
4              8.30       22186    0.00   0.00     0.00     0.00      0.00   
...             ...         ...     ...    ...      ...      ...       ...   
26512          0.00           1    0.00   1.00     0.00     0.00      0.00   
26513          0.00           1    0.00   1.00     0.00     0.00      1.00   
26514          0.00           1    0.00   0.00     0.00     0.00      0.00   
26515          0.00           1    0.00   0.00     0.00     0.00      0.00   
26516          0.00           1    0.00   0.00     0.00     0.00      1.00   

       Romance  History  Horror  Fantasy  Music  Adventure  Documentary  \
0         0.00     0.00    0.00     1.00   0.00       1.00         0.00   
1         0.00     0.00    0.00     1.00   0.00       1.00         0.00   
2         0.00     0.00    0.00     0.00   0.00       1.00         0.00   
3         0.00     0.00    0.00     0.00   0.00       0.00         0.00   
4         0.00     0.00    0.00     0.00   0.00       1.00         0.00   
...        ...      ...     ...      ...    ...        ...          ...   
26512     0.00     0.00    1.00     0.00   0.00       0.00         0.00   
26513     0.00     0.00    0.00     0.00   0.00       0.00         0.00   
26514     0.00     0.00    0.00     1.00   0.00       1.00         0.00   
26515     0.00     0.00    0.00     0.00   0.00       1.00         0.00   
26516     0.00     0.00    1.00     0.00   0.00       0.00         0.00   

       Family  War  Crime  Science Fiction  Animation  Action  TV Movie  \
0        1.00 0.00   0.00             0.00       0.00    0.00      0.00   
1        1.00 0.00   0.00             0.00       1.00    0.00      0.00   
2        0.00 0.00   0.00             1.00       0.00    1.00      0.00   
3      

In [41]:
tmdb_lang_df = tmdb_lang_df.drop_duplicates(subset='original_title', keep='first')
tmdb_lang_df

Unnamed: 0      id                                original_title  \
0               0   12444  Harry Potter and the Deathly Hallows: Part 1   
1               1   10191                      How to Train Your Dragon   
2               2   10138                                    Iron Man 2   
3               3     862                                     Toy Story   
4               4   27205                                     Inception   
...           ...     ...                                           ...   
26512       26512  488143                         Laboratory Conditions   
26513       26513  485975                               _EXHIBIT_84xxx_   
26514       26514  381231                                  The Last One   
26515       26515  366854                                  Trailer Made   
26516       26516  309885                                    The Church   

       popularity release_date                                         title  \
0           33.53   2010-11-19  Harry Potter and the Deathly Hallows: Part 1   
1           28.73   2010-03-26                      How to Train Your Dragon   
2           28.52   2010-05-07                                    Iron Man 2   
3           28.00   1995-11-22                                     Toy Story   
4           27.92   2010-07-16                                     Inception   
...           ...          ...                                           ...   
26512        0.60   2018-10-13                         Laboratory Conditions   
26513        0.60   2018-05-01                               _EXHIBIT_84xxx_   
26514        0.60   2018-10-01                                  The Last One   
26515        0.60   2018-06-22                                  Trailer Made   
26516        0.60   2018-10-05                                    The Church   

       vote_average  vote_count  Comedy  Drama  Mystery  Western  Thriller  \
0              7.70       10788    0.00   0.00     0.00     0.00      0.00   
1              7.70        7610    0.00   0.00     0.00     0.00      0.00   
2              6.80       12368    0.00   0.00     0.00     0.00      0.00   
3              7.90       10174    1.00   0.00     0.00     0.00      0.00   
4              8.30       22186    0.00   0.00     0.00     0.00      0.00   
...             ...         ...     ...    ...      ...      ...       ...   
26512          0.00           1    0.00   1.00     0.00     0.00      0.00   
26513          0.00           1    0.00   1.00     0.00     0.00      1.00   
26514          0.00           1    0.00   0.00     0.00     0.00      0.00   
26515          0.00           1    0.00   0.00     0.00     0.00      0.00   
26516          0.00           1    0.00   0.00     0.00     0.00      1.00   

       Romance  History  Horror  Fantasy  Music  Adventure  Documentary  \
0         0.00     0.00    0.00     1.00   0.00       1.00         0.00   
1         0.00     0.00    0.00     1.00   0.00       1.00         0.00   
2         0.00     0.00    0.00     0.00   0.00       1.00         0.00   
3         0.00     0.00    0.00     0.00   0.00       0.00         0.00   
4         0.00     0.00    0.00     0.00   0.00       1.00         0.00   
...        ...      ...     ...      ...    ...        ...          ...   
26512     0.00     0.00    1.00     0.00   0.00       0.00         0.00   
26513     0.00     0.00    0.00     0.00   0.00       0.00         0.00   
26514     0.00     0.00    0.00     1.00   0.00       1.00         0.00   
26515     0.00     0.00    0.00     0.00   0.00       1.00         0.00   
26516     0.00     0.00    1.00     0.00   0.00       0.00         0.00   

       Family  War  Crime  Science Fiction  Animation  Action  TV Movie  \
0        1.00 0.00   0.00             0.00       0.00    0.00      0.00   
1        1.00 0.00   0.00             0.00       1.00    0.00      0.00   
2        0.00 0.00   0.00             1.00       0.00    1.00      0.00   
3      

In [42]:
tmdb_lang_df = tmdb_lang_df.dropna()
tmdb_lang_df

Unnamed: 0      id                                original_title  \
0               0   12444  Harry Potter and the Deathly Hallows: Part 1   
1               1   10191                      How to Train Your Dragon   
2               2   10138                                    Iron Man 2   
3               3     862                                     Toy Story   
4               4   27205                                     Inception   
...           ...     ...                                           ...   
26512       26512  488143                         Laboratory Conditions   
26513       26513  485975                               _EXHIBIT_84xxx_   
26514       26514  381231                                  The Last One   
26515       26515  366854                                  Trailer Made   
26516       26516  309885                                    The Church   

       popularity release_date                                         title  \
0           33.53   2010-11-19  Harry Potter and the Deathly Hallows: Part 1   
1           28.73   2010-03-26                      How to Train Your Dragon   
2           28.52   2010-05-07                                    Iron Man 2   
3           28.00   1995-11-22                                     Toy Story   
4           27.92   2010-07-16                                     Inception   
...           ...          ...                                           ...   
26512        0.60   2018-10-13                         Laboratory Conditions   
26513        0.60   2018-05-01                               _EXHIBIT_84xxx_   
26514        0.60   2018-10-01                                  The Last One   
26515        0.60   2018-06-22                                  Trailer Made   
26516        0.60   2018-10-05                                    The Church   

       vote_average  vote_count  Comedy  Drama  Mystery  Western  Thriller  \
0              7.70       10788    0.00   0.00     0.00     0.00      0.00   
1              7.70        7610    0.00   0.00     0.00     0.00      0.00   
2              6.80       12368    0.00   0.00     0.00     0.00      0.00   
3              7.90       10174    1.00   0.00     0.00     0.00      0.00   
4              8.30       22186    0.00   0.00     0.00     0.00      0.00   
...             ...         ...     ...    ...      ...      ...       ...   
26512          0.00           1    0.00   1.00     0.00     0.00      0.00   
26513          0.00           1    0.00   1.00     0.00     0.00      1.00   
26514          0.00           1    0.00   0.00     0.00     0.00      0.00   
26515          0.00           1    0.00   0.00     0.00     0.00      0.00   
26516          0.00           1    0.00   0.00     0.00     0.00      1.00   

       Romance  History  Horror  Fantasy  Music  Adventure  Documentary  \
0         0.00     0.00    0.00     1.00   0.00       1.00         0.00   
1         0.00     0.00    0.00     1.00   0.00       1.00         0.00   
2         0.00     0.00    0.00     0.00   0.00       1.00         0.00   
3         0.00     0.00    0.00     0.00   0.00       0.00         0.00   
4         0.00     0.00    0.00     0.00   0.00       1.00         0.00   
...        ...      ...     ...      ...    ...        ...          ...   
26512     0.00     0.00    1.00     0.00   0.00       0.00         0.00   
26513     0.00     0.00    0.00     0.00   0.00       0.00         0.00   
26514     0.00     0.00    0.00     1.00   0.00       1.00         0.00   
26515     0.00     0.00    0.00     0.00   0.00       1.00         0.00   
26516     0.00     0.00    1.00     0.00   0.00       0.00         0.00   

       Family  War  Crime  Science Fiction  Animation  Action  TV Movie  \
0        1.00 0.00   0.00             0.00       0.00    0.00      0.00   
1        1.00 0.00   0.00             0.00       1.00    0.00      0.00   
2        0.00 0.00   0.00             1.00       0.00    1.00      0.00   
3      

In [43]:
tmdb_all_lang_df = tmdb_lang_df.loc[:, ['original_title', 'original_language', 'release_date', 'popularity']]
tmdb_all_lang_df

original_title original_language  \
0      Harry Potter and the Deathly Hallows: Part 1           English   
1                          How to Train Your Dragon           English   
2                                        Iron Man 2           English   
3                                         Toy Story           English   
4                                         Inception           English   
...                                             ...               ...   
26512                         Laboratory Conditions           English   
26513                               _EXHIBIT_84xxx_           English   
26514                                  The Last One           English   
26515                                  Trailer Made           English   
26516                                    The Church           English   

      release_date  popularity  
0       2010-11-19       33.53  
1       2010-03-26       28.73  
2       2010-05-07       28.52  
3       1995-11-22       28.00  
4       2010-07-16       27.92  
...            ...         ...  
26512   2018-10-13        0.60  
26513   2018-05-01        0.60  
26514   2018-10-01        0.60  
26515   2018-06-22        0.60  
26516   2018-10-05        0.60  

[24835 rows x 4 columns]

In [44]:
tmdblangne = tmdb_all_lang_df[~tmdb_all_lang_df['original_language'].isin(['English'])] #filter out 'English' value in 'original_language' col
tmdblangne

original_title original_language release_date  popularity
17                     LelleBelle             Dutch   2010-10-09       18.59
49     Tres metros sobre el cielo           Spanish   2010-12-20       13.72
70         El secreto de sus ojos           Spanish   2010-04-16       12.53
75                        サマーウォーズ          Japanese   2010-10-13       12.28
79      Luftslottet som sprängdes           Swedish   2010-10-29       12.23
...                           ...               ...          ...         ...
26457           Maison du bonheur            French   2018-08-24        0.60
26469          Point of No Return        Portuguese   2018-09-19        0.60
26494            La última virgen           Spanish   2018-05-26        0.60
26503                Evolution 4K            German   2018-06-12        0.60
26507                   Dreamaway            Arabic   2018-10-14        0.60

[3079 rows x 4 columns]

21, 756 English as original language vs. 3079 non-English language films in dataset. 

In [45]:
tmdblangnesort = tmdblangne.groupby('original_language').count()
tmdblangnesort = tmdblangnesort.sort_values(by=['original_title'], ascending=False)
tmdblangnesort = tmdblangnesort.reset_index()
tmdblangnesort

original_language  original_title  release_date  popularity
0             French             479           479         479
1            Spanish             432           432         432
2            Russian             295           295         295
3           Japanese             243           243         243
4             German             224           224         224
5           Mandarin             174           174         174
6              Hindi             168           168         168
7            Italian             118           118         118
8         Portuguese              93            93          93
9             Korean              92            92          92
10           Swedish              65            65          65
11            Polish              50            50          50
12            Danish              50            50          50
13         Norwegian              48            48          48
14         Cantonese              48            48          48
15             Dutch              46            46          46
16            Hebrew              37            37          37
17              Thai              36            36          36
18            Arabic              31            31          31
19            Telugu              30            30          30
20           Persian              26            26          26
21       No Language              24            24          24
22           Turkish              19            19          19
23          Romanian              19            19          19
24         Hungarian              18            18          18
25             Greek              15            15          15
26           Tagalog              15            15          15
27           Finnish              14            14          14
28        Indonesian              13            13          13
29              Urdu              11            11          11
30         Icelandic               9             9           9
31             Czech               9             9           9
32         Malayalam               9             9           9
33           Kannada               7             7           7
34          Georgian               7             7           7
35           Serbian               7             7           7
36        Vietnamese               7             7           7
37           Latvian               6             6           6
38        Lithuanian               6             6           6
39             Tamil               6             6           6
40         Ukrainian               6             6           6
41           Marathi               5             5           5
42          Estonian               5             5           5
43         Bulgarian               4             4           4
44           Bengali               4             4           4
45         Abkhazian               4             4           4
46           Punjabi               3             3           3
47           Kurdish               3             3           3
48               Lao               3             3           3
49          Albanian               3             3           3
50           Catalan               3             3           3
51          Croatian               3             3           3
52            Basque               2             2           2
53           Bosnian               2             2           2
54            Nepali               2             2           2
55          Gujarati               2             2           2
56  Norwegian Bokmål               1             1           1
57          Dzongkha               1             1           1
58             Xhosa               1             1           1
59             Welsh               1             1           1
60          Armenian               1             1           1
61           Tibetan               1             1           1
62              Cree              

In [49]:
tn_movie_budgets_df

release_date                                        movie  \
id                                                              
1   Dec 18, 2009                                       Avatar   
2   May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
3    Jun 7, 2019                                 Dark Phoenix   
4    May 1, 2015                      Avengers: Age of Ultron   
5   Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   
..           ...                                          ...   
78  Dec 31, 2018                                       Red 11   
79   Apr 2, 1999                                    Following   
80  Jul 13, 2005                Return to the Land of Wonders   
81  Sep 29, 2015                         A Plague So Pleasant   
82   Aug 5, 2005                            My Date With Drew   

    production_budget  domestic_gross  worldwide_gross  
id                                                      
1           425000000       760507625       2776345279  
2           410600000       241063875       1045663875  
3           350000000        42762350        149762350  
4           330600000       459005868       1403013963  
5           317000000       620181382       1316721747  
..                ...             ...              ...  
78               7000               0                0  
79               6000           48482           240495  
80               5000            1338             1338  
81               1400               0                0  
82               1100          181041           181041  

[5782 rows x 5 columns]

In [51]:
tn_movie_budgets_df['foreign_gross'] = tn_movie_budgets_df['worldwide_gross'] - tn_movie_budgets_df['domestic_gross']
tn_movie_budgets_df

release_date                                        movie  \
id                                                              
1   Dec 18, 2009                                       Avatar   
2   May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
3    Jun 7, 2019                                 Dark Phoenix   
4    May 1, 2015                      Avengers: Age of Ultron   
5   Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   
..           ...                                          ...   
78  Dec 31, 2018                                       Red 11   
79   Apr 2, 1999                                    Following   
80  Jul 13, 2005                Return to the Land of Wonders   
81  Sep 29, 2015                         A Plague So Pleasant   
82   Aug 5, 2005                            My Date With Drew   

    production_budget  domestic_gross  worldwide_gross  foreign_gross  
id                                                                     
1           425000000       760507625       2776345279     2015837654  
2           410600000       241063875       1045663875      804600000  
3           350000000        42762350        149762350      107000000  
4           330600000       459005868       1403013963      944008095  
5           317000000       620181382       1316721747      696540365  
..                ...             ...              ...            ...  
78               7000               0                0              0  
79               6000           48482           240495         192013  
80               5000            1338             1338              0  
81               1400               0                0              0  
82               1100          181041           181041              0  

[5782 rows x 6 columns]

In [ ]:
tn_movie_budgets_df = tn_movie_budgets_df.sort_values(by=['worldwide_gross'], ascending=False)
tn_movie_budgets_df = tn_movie_budgets_df.reset_index(drop=True)
top10grosseng = tn_movie_budgets_df.loc[:9,['movie','worldwide_gross']]
top10grosseng

In [57]:
tn_movie_budgets_df = tn_movie_budgets_df.sort_values(by=['foreign_gross'], ascending=False)
tn_movie_budgets_df = tn_movie_budgets_df.reset_index(drop=True)
top10foreng = tn_movie_budgets_df.loc[:9,['movie','foreign_gross']]
top10foreng

movie  foreign_gross
0                                         Avatar     2015837654
1                                        Titanic     1548844451
2                         Avengers: Infinity War     1369318718
3                                      Furious 7     1165715774
4           Star Wars Ep. VII: The Force Awakens     1116648995
5                        The Fate of the Furious     1009081502
6                                 Jurassic World      996584239
7  Harry Potter and the Deathly Hallows: Part II      960500000
8                        Avengers: Age of Ultron      944008095
9                                   The Avengers      894656350

In [58]:
top10foreng.loc[:,'foreign_gross'].sum()

12021195778